src/ucs/memory/memtype_cache.c

```diff
@@ -27,11 +27,25 @@ ucs_spinlock_t ucs_memtype_cache_global_instance_lock;
 ucs_memtype_cache_t *ucs_memtype_cache_global_instance = NULL;
 
 
+#define UCS_MEMTYPE_CACHE_REGION_FMT UCS_PGT_REGION_FMT " %s dev %s"
+#define UCS_MEMTYPE_CACHE_REGION_ARG(_region) \
+            UCS_PGT_REGION_ARG(&(_region)->super), \
+            ucs_memory_type_names[(_region)->mem_type], \
+            ucs_topo_sys_device_get_name((_region)->sys_dev)
+
 typedef enum {
     UCS_MEMTYPE_CACHE_ACTION_SET_MEMTYPE,
     UCS_MEMTYPE_CACHE_ACTION_REMOVE
 } ucs_memtype_cache_action_t;
 
+struct ucs_memtype_cache_region {
+    ucs_pgt_region_t  super;    /**< Base class - page table region */
+    ucs_list_link_t   list;     /**< List element */
+    ucs_memory_type_t mem_type; /**< Memory type, use uint8 for compact size */
```

Comment:
was the intent to use `uint8_t`?

Reply:
no, since it's replacing ucs_memory_info_t which has ucs_memory_type_t

Reply:
then the comment is misleading

Reply:
yeah :)